File: ed2/gara2/negozi/att/negozi.c
#include <stdio.h>
#include <assert.h>

#define MAXN 100000
#define MAXQ 100000

void risolvi(int N, int Q, int t[], int a[], int b[], int d[]) {
    // Mettete qui il codice della soluzione
    int i;
    for (i=0; i<Q; i++) {
        d[i] = 42;
    }
}


int t[MAXN], a[MAXQ], b[MAXQ], d[MAXQ];

int main() {
    FILE *fr, *fw;
    int N, Q, i;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(2 == fscanf(fr, "%d%d", &N, &Q));
    for(i=0; i<N; i++)
        assert(1 == fscanf(fr, "%d", &t[i]));
    for(i=0; i<Q; i++)
        assert(2 == fscanf(fr, "%d%d", &a[i], &b[i]));

    risolvi(N, Q, t, a, b, d);

    for(i=0; i<Q; i++)
        fprintf(fw, "%d\n", d[i]);
    fclose(fr);
    fclose(fw);
    return 0;
}

File: ed2/gara2/negozi/att/negozi.pas
procedure risolvi(N, Q: longint; var t, a, b, d: array of longint);
var i: longint;
begin
    (* Mettete qui il codice della soluzione *)
    for i:=0 to Q-1 do
        d[i] := 42;
end;


const
    MAXN = 100000;
    MAXQ = 100000;
var
    N, Q, i  : longint;
    t        : array[0..MAXN-1] of longint;
    a, b, d  : array[0..MAXQ-1] of longint;
    fr, fw: text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, N, Q);
    for i:=0 to N-1 do
        read(fr, t[i]);
    readln;
    for i:=0 to Q-1 do
        readln(fr, a[i], b[i]);

    risolvi(N, Q, t, a, b, d);

    for i:=0 to Q-1 do
        writeln(fw, d[i]);
    close(fr);
    close(fw);
end.
